### ISSUE TITLE
Unable to compile on Mac OS X 10.11.6

### ISSUE THREAD

**Issue by dplummer** [skunkwerks/snappy-erlang-nif#18]
```
$ rebar compile
==> snappy-erlang-nif (compile)
cc /Users/dplummer/src/elixir/snappy-erlang-nif/c_src/snappy/snappy-sinksource.o /Users/dplummer/src/elixir/snappy-erlang-nif/c_src/snappy/snappy-stubs-internal.o /Users/dplummer/src/elixir/snappy-erlang-nif/c_src/snappy/snappy.o /Users/dplummer/src/elixir/snappy-erlang-nif/c_src/snappy_nif.o -L/usr/local/lib -shared -lstdc++ -L /usr/local/Cellar/erlang/19.0.2/lib/erlang/lib/erl_interface-3.9/lib -lerl_interface -lei -o /Users/dplummer/src/elixir/snappy-erlang-nif/c_src/../priv/snappy_nif.so
Undefined symbols for architecture x86_64:
  "_enif_alloc_binary", referenced from:
      SnappyNifSink::SnappyNifSink(enif_environment_t*) in snappy_nif.o
      SnappyNifSink::SnappyNifSink(enif_environment_t*) in snappy_nif.o
      _snappy_compress_erl in snappy_nif.o
      _snappy_decompress_erl in snappy_nif.o
  "_enif_inspect_iolist_as_binary", referenced from:
      _snappy_compress_erl in snappy_nif.o
      _snappy_decompress_erl in snappy_nif.o
      _snappy_uncompressed_length_erl in snappy_nif.o
      _snappy_is_valid in snappy_nif.o
  "_enif_make_atom", referenced from:
      _snappy_compress_erl in snappy_nif.o
      _snappy_decompress_erl in snappy_nif.o
      _snappy_uncompressed_length_erl in snappy_nif.o
      _snappy_is_valid in snappy_nif.o
  "_enif_make_badarg", referenced from:
      _snappy_compress_erl in snappy_nif.o
      _snappy_decompress_erl in snappy_nif.o
      _snappy_uncompressed_length_erl in snappy_nif.o
      _snappy_is_valid in snappy_nif.o
  "_enif_make_binary", referenced from:
      _snappy_compress_erl in snappy_nif.o
      _snappy_decompress_erl in snappy_nif.o
  "_enif_make_existing_atom", referenced from:
      _snappy_compress_erl in snappy_nif.o
      _snappy_decompress_erl in snappy_nif.o
      _snappy_uncompressed_length_erl in snappy_nif.o
      _snappy_is_valid in snappy_nif.o
  "_enif_make_tuple", referenced from:
      _snappy_compress_erl in snappy_nif.o
      _snappy_decompress_erl in snappy_nif.o
      _snappy_uncompressed_length_erl in snappy_nif.o
      _snappy_is_valid in snappy_nif.o
  "_enif_make_ulong", referenced from:
      _snappy_uncompressed_length_erl in snappy_nif.o
  "_enif_realloc_binary", referenced from:
      SnappyNifSink::GetAppendBuffer(unsigned long, char*) in snappy_nif.o
      SnappyNifSink::getBin() in snappy_nif.o
      _snappy_compress_erl in snappy_nif.o
  "_enif_release_binary", referenced from:
      SnappyNifSink::~SnappyNifSink() in snappy_nif.o
      SnappyNifSink::~SnappyNifSink() in snappy_nif.o
      SnappyNifSink::~SnappyNifSink() in snappy_nif.o
      _snappy_compress_erl in snappy_nif.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [/Users/dplummer/src/elixir/snappy-erlang-nif/c_src/../priv/snappy_nif.so] Error 1
ERROR: Command [compile] failed!
```

`erl -v`:

```
Erlang/OTP 19 [erts-8.0.2] [source] [64-bit] [smp:8:8] [async-threads:10] [hipe] [kernel-poll:false] [dtrace]
```

`gcc -v`:

```
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 7.3.0 (clang-703.0.31)
Target: x86_64-apple-darwin15.6.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

Ideas? I'm pretty new to erlang, and know basically nothing of nifs. Just trying to use this library with [kafka_ex](https://github.com/kafkaex/kafka_ex)

**Comment by dch** [comment]
can you try `./rebar compile -vvv` and paste the output in here please? Make sure you clean any passwords/tokens from the output - environment variables are pulled in.

I have the same XCode/erl setup but don't have the same issue.

**Comment by dplummer** [comment]
Sorry for the late response. Here's what I get for running `./rebar compile -vvv`

```
DEBUG: Rebar location: "/Users/dplummer/src/elixir/snappy-erlang-nif/rebar"
DEBUG: Evaluating config script "/Users/dplummer/src/elixir/snappy-erlang-nif/rebar.config.script"
DEBUG: Consult config file "/Users/dplummer/src/elixir/snappy-erlang-nif/rebar.config"
DEBUG: Available deps: []
DEBUG: Missing deps  : []
DEBUG: Predirs: []
==> snappy-erlang-nif (compile)
INFO:  sh info:
	cwd: "/Users/dplummer/src/elixir/snappy-erlang-nif"
	cmd: make -C c_src
DEBUG: 	opts: [{env,[{"REBAR_DEPS_DIR",
                      "/Users/dplummer/src/elixir/snappy-erlang-nif/deps"},
                     {"ERL_LIBS",
                      "/Users/dplummer/src/elixir/snappy-erlang-nif/deps"},
                     {"AUTOFEATURE","true"},
                     {"AWS_AUTO_SCALING_HOME",
                      "/home/dplummer/src/ec2-autoscaling/AutoScaling-1.0.49.1"},
                     {"Apple_PubSub_Socket_Render",
                      "/private/tmp/com.apple.launchd.fmhCHHbcsV/Render"},
                     {"BINDIR",
                      "/usr/local/Cellar/erlang/19.1/lib/erlang/erts-8.1/bin"},
                     {"BROWSER","google-chrome"},
                     {"CC","cc"},
                     {"COLORFGBG","7;0"},
                     {"COMMAND_MODE","unix2003"},
                     {"CPLUS_INCLUDE_PATH","/usr/local/opt/openssl/include"},
                     {"CXX","c++"},
                     {"DEFAULT_USER","dplummer"},
                     {"DISPLAY",":0"},
                     {"DRV_CC_TEMPLATE",
                      "cc -c $CFLAGS -g -Wall -fPIC  -I/usr/local/Cellar/erlang/19.1/lib/erlang/lib/erl_interface-3.9.1/include -I/usr/local/Cellar/erlang/19.1/lib/erlang/erts-8.1/include   $PORT_IN_FILES -o $PORT_OUT_FILE"},
                     {"DRV_CFLAGS",
                      "-g -Wall -fPIC  -I/usr/local/Cellar/erlang/19.1/lib/erlang/lib/erl_interface-3.9.1/include -I/usr/local/Cellar/erlang/19.1/lib/erlang/erts-8.1/include  "},
                     {"DRV_CXX_TEMPLATE",
                      "c++ -c $CXXFLAGS -g -Wall -fPIC  -I/usr/local/Cellar/erlang/19.1/lib/erlang/lib/erl_interface-3.9.1/include -I/usr/local/Cellar/erlang/19.1/lib/erlang/erts-8.1/include   $PORT_IN_FILES -o $PORT_OUT_FILE"},
                     {"DRV_LDFLAGS",
                      "-bundle -flat_namespace -undefined suppress  -L/usr/local/Cellar/erlang/19.1/lib/erlang/lib/erl_interface-3.9.1/lib -lerl_interface -lei"},
                     {"DRV_LINK_TEMPLATE",
                      "cc $PORT_IN_FILES -L/usr/local/lib -bundle -flat_namespace -undefined suppress  -L/usr/local/Cellar/erlang/19.1/lib/erlang/lib/erl_interface-3.9.1/lib -lerl_interface -lei -o $PORT_OUT_FILE"},
                     {"EDITOR","vim"},
                     {"EMU","beam"},
                     {"ERLANG_ARCH","64"},
                     {"ERLANG_TARGET","19-x86_64-apple-darwin16.0.0-64"},
                     {"ERL_CFLAGS",
                      " -I/usr/local/Cellar/erlang/19.1/lib/erlang/lib/erl_interface-3.9.1/include -I/usr/local/Cellar/erlang/19.1/lib/erlang/erts-8.1/include  "},
                     {"ERL_EI_LIBDIR",
                      "/usr/local/Cellar/erlang/19.1/lib/erlang/lib/erl_interface-3.9.1/lib"},
                     {"ERL_LDFLAGS",
                      " -L/usr/local/Cellar/erlang/19.1/lib/erlang/lib/erl_interface-3.9.1/lib -lerl_interface -lei"},
                     {"EXE_CC_TEMPLATE",
                      "cc -c $CFLAGS -g -Wall -fPIC  -I/usr/local/Cellar/erlang/19.1/lib/erlang/lib/erl_interface-3.9.1/include -I/usr/local/Cellar/erlang/19.1/lib/erlang/erts-8.1/include   $PORT_IN_FILES -o $PORT_OUT_FILE"},
                     {"EXE_CFLAGS",
                      "-g -Wall -fPIC  -I/usr/local/Cellar/erlang/19.1/lib/erlang/lib/erl_interface-3.9.1/include -I/usr/local/Cellar/erlang/19.1/lib/erlang/erts-8.1/include  "},
                     {"EXE_CXX_TEMPLATE",
                      "c++ -c $CXXFLAGS -g -Wall -fPIC  -I/usr/local/Cellar/erlang/19.1/lib/erlang/lib/erl_interface-3.9.1/include -I/usr/local/Cellar/erlang/19.1/lib/erlang/erts-8.1/include   $PORT_IN_FILES -o $PORT_OUT_FILE"},
                     {"EXE_LDFLAGS",
                      " -L/usr/local/Cellar/erlang/19.1/lib/erlang/lib/erl_interface-3.9.1/lib -lerl_interface -lei"},
                     {"EXE_LINK_TEMPLATE",
                      "cc $PORT_IN_FILES -L/usr/local/lib  -L/usr/local/Cellar/erlang/19.1/lib/erlang/lib/erl_interface-3.9.1/lib -lerl_interface -lei -o $PORT_OUT_FILE"},
                     {"GIT_EDITOR","vim -f"},
                     {"GOPATH","/Users/dplummer"},
                     {"GOROOT","/usr/local/Cellar/go/1.3.1/libexec"},
                     {"HOME","/Users/dplummer"},
                     {"ITERM_PROFILE","Default"},
                     {"ITERM_SHELL_INTEGRATION_INSTALLED","Yes"},
                     {"KAFKA_HOSTS","localhost:9092"},
                     {"LANG","en_US.UTF-8"},
                     {"LC_CTYPE","en_US.UTF-8"},
                     {"LDFLAGS","-L/usr/local/lib"},
                     {"LESS","-R"},
                     {"LOGNAME","dplummer"},
                     {"LSCOLORS","Gxfxcxdxbxegedabagacad"},
                     {"OBJCPLUS_INCLUDE_PATH","/usr/local/opt/openssl/lib"},
                     {"PAGER","less"},
                     {"PATH",
                      "/usr/local/Cellar/erlang/19.1/lib/erlang/erts-8.1/bin:/usr/local/Cellar/erlang/19.1/lib/erlang/bin:/Users/dplummer/.rbenv/shims:/Users/dplummer/bin:/usr/local/bin:/opt/local/bin:/opt/local/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/TeX/texbin:/usr/local/Cellar/mysql/5.1.54/bin:/Users/dplummer/bin:/Users/dplummer/.cabal/bin:/opt/java/jre/bin:/opt/ruby-enterprise-1.8.7-2011.03/bin:/home/dplummer/src/ec2-autoscaling/AutoScaling-1.0.49.1/bin:/home/dplummer/bin:/opt/android-sdk/tools:/Users/dplummer/bin:/usr/local/opt/go/libexec/bin"},
                     {"PROGNAME","erl"},
                     {"PWD","/Users/dplummer/src/elixir/snappy-erlang-nif"},
                     {"RBENV_SHELL","zsh"},
                     {"ROOTDIR","/usr/local/Cellar/erlang/19.1/lib/erlang"},
                     {"RUBY_GC_MALLOC_LIMIT","200000000"},
                     {"RUBY_HEAP_SLOTS_GROWTH_FACTOR","1"},
                     {"RUBY_HEAP_SLOTS_INCREMENT","1000000"},
                     {"SHELL","/bin/zsh"},
                     {"SHLVL","1"},
                     {"SSL_CERT_FILE",
                      "/Library/Java/JavaVirtualMachines/jdk1.8.0_65.jdk/Contents/Home/jre/lib/security/cacerts"},
                     {"STDOUT_SYNC","true"},
                     {"TERM","xterm-256color"},
                     {"TERM_PROGRAM","iTerm.app"},
                     {"TERM_PROGRAM_VERSION","3.0.12"},
                     {"TERM_SESSION_ID",
                      "w0t2p0:30A4F729-86A5-44BF-941C-4AE2B4E04FA2"},
                     {"TMPDIR",
                      "/var/folders/77/rxrpzsrs0pn4wkhj9w5hnggc0000gn/T/"},
                     {"USER","dplummer"},
                     {"XPC_FLAGS","0x0"},
                     {"XPC_SERVICE_NAME","0"},
                     {"ZSH","/Users/dplummer/.oh-my-zsh"},
                     {"ZSH_THEME","agnoster"},
                     {"__CF_USER_TEXT_ENCODING","0x1F5:0x0:0x0"}]},
               {abort_on_error,"Command [compile] failed!\n"}]
cc /Users/dplummer/src/elixir/snappy-erlang-nif/c_src/snappy/snappy-sinksource.o /Users/dplummer/src/elixir/snappy-erlang-nif/c_src/snappy/snappy-stubs-internal.o /Users/dplummer/src/elixir/snappy-erlang-nif/c_src/snappy/snappy.o /Users/dplummer/src/elixir/snappy-erlang-nif/c_src/snappy_nif.o -L/usr/local/lib -shared -lstdc++ -L /usr/local/Cellar/erlang/19.1/lib/erlang/lib/erl_interface-3.9.1/lib -lerl_interface -lei -o /Users/dplummer/src/elixir/snappy-erlang-nif/c_src/../priv/snappy_nif.so
Undefined symbols for architecture x86_64:
  "_enif_alloc_binary", referenced from:
      SnappyNifSink::SnappyNifSink(enif_environment_t*) in snappy_nif.o
      SnappyNifSink::SnappyNifSink(enif_environment_t*) in snappy_nif.o
      _snappy_compress_erl in snappy_nif.o
      _snappy_decompress_erl in snappy_nif.o
  "_enif_inspect_iolist_as_binary", referenced from:
      _snappy_compress_erl in snappy_nif.o
      _snappy_decompress_erl in snappy_nif.o
      _snappy_uncompressed_length_erl in snappy_nif.o
      _snappy_is_valid in snappy_nif.o
  "_enif_make_atom", referenced from:
      _snappy_compress_erl in snappy_nif.o
      _snappy_decompress_erl in snappy_nif.o
      _snappy_uncompressed_length_erl in snappy_nif.o
      _snappy_is_valid in snappy_nif.o
  "_enif_make_badarg", referenced from:
      _snappy_compress_erl in snappy_nif.o
      _snappy_decompress_erl in snappy_nif.o
      _snappy_uncompressed_length_erl in snappy_nif.o
      _snappy_is_valid in snappy_nif.o
  "_enif_make_binary", referenced from:
      _snappy_compress_erl in snappy_nif.o
      _snappy_decompress_erl in snappy_nif.o
  "_enif_make_existing_atom", referenced from:
      _snappy_compress_erl in snappy_nif.o
      _snappy_decompress_erl in snappy_nif.o
      _snappy_uncompressed_length_erl in snappy_nif.o
      _snappy_is_valid in snappy_nif.o
  "_enif_make_tuple", referenced from:
      _snappy_compress_erl in snappy_nif.o
      _snappy_decompress_erl in snappy_nif.o
      _snappy_uncompressed_length_erl in snappy_nif.o
      _snappy_is_valid in snappy_nif.o
  "_enif_make_ulong", referenced from:
      _snappy_uncompressed_length_erl in snappy_nif.o
  "_enif_realloc_binary", referenced from:
      SnappyNifSink::GetAppendBuffer(unsigned long, char*) in snappy_nif.o
      SnappyNifSink::getBin() in snappy_nif.o
      _snappy_compress_erl in snappy_nif.o
  "_enif_release_binary", referenced from:
      SnappyNifSink::~SnappyNifSink() in snappy_nif.o
      SnappyNifSink::~SnappyNifSink() in snappy_nif.o
      SnappyNifSink::~SnappyNifSink() in snappy_nif.o
      _snappy_compress_erl in snappy_nif.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [/Users/dplummer/src/elixir/snappy-erlang-nif/c_src/../priv/snappy_nif.so] Error 1
ERROR: Command [compile] failed!
```

**Comment by dch** [comment]
The only thing I can guess at is that perhaps you already have a snappy library in /usr/local/lib (LDFLAGS) that's getting pulled in. Otherwise, there's nothing else different to here.

Can you make sure you've got no snappy*.so lying around (perhaps another brew recipe)?

**Comment by dtabwhite** [comment]
I just ran into this while trying to use this in kafka_ex as well and I believe the issue happens if you already have LDFLAGS set in our env.  The issue went away for me when I `unset LDFLAGS` in my (bash) shell.  Looks like these are conditionally set over here: https://github.com/skunkwerks/snappy-erlang-nif/blob/master/c_src/Makefile#L22

**Comment by dch** [comment]
closed via #26 thanks for reporting the work-around
